### DISCUSSION TITLE
Why `setlocal scrolloff` changes after opening another buffer in the same window?

### DISCUSSION THREAD

**Discussion by przepompownia** [neovim/neovim#31718] [Q&A] [answered]
For example
```
nvim --clean -c 'setlocal scrolloff=10' f1 -c 'edit f2' -c 'set scrolloff'
```
shows 0. 

If I understand correctly, the local `scrolloff` value is bound to the current window that does not change on `edit`. Does the first buffer also restrict the `scrolloff` scope?

I tried to explicitly bound that local option to the only window but it get lost after changing buffer
```
nvim-src-clean -c 'lua vim.wo[1000].scrolloff = 10' f1 -c 'edit f2' -c 'set scrolloff'
```
also with no result.

How to keep such local option attached to a window regardless of buffer and without autocommands?

**Comment by zeertzjq** [comment, marked as answer]
See `:h local-options`

**Reply by przepompownia** [reply]
> The options local to a window are remembered for each buffer.

seems to confirm that. In the following example

```sh
nvim --clean -c 'setl so=10' f1 -c 'vnew' -c 'e f1' -c 'se so'
```
the new window gets `so` value from `f1` buffer, but we can still set different local `so` values for both windows later. 

Thanks for the docs.
